src/update.js

```diff
@@ -0,0 +1,137 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const fs = require('fs');
+const path = require('path');
+const readline = require('readline');
+const c = require('./config');
+
+/**
+ * Update the project and user npmrc files.
+ *
+ * @param {string} fromConfigPath Path to the npmrc file to read scope registry configs from, should be the project npmrc file.
+ * @param {string} toConfigPath Path to npmrc file to write authentication configs to, should be the user npmrc file.
+ * @param {string} creds Encrypted credentials.
+ * @return {!Promise<undefined>}
+ */
+async function updateConfigFiles(fromConfigPath, toConfigPath, creds) {
+  fromConfigPath = path.resolve(fromConfigPath);
+  toConfigPath = path.resolve(toConfigPath);
+  // Backward-compatible scenario. Update auth configs in project npmrc directly.
+  if (fromConfigPath == toConfigPath) {
+    updateConfigFile(fromConfigPath, creds);
+    return;
+  }
+
+  let fromConfigs = new Array();
+  let toConfigs = new Array();
+  let registryAuthConfigs = new Map();
+
+  // We do not use basic auth any more in `gcloud artifacts print-settings`; replace them.
+  let fromConfigLines = await fs.promises.readFile(fromConfigPath, "utf8")
+  const legacyRegex = /(\/\/[a-zA-Z1-9-]+[-]npm[.]pkg[.]dev\/.*\/):_password=.*(\n\/\/[a-zA-Z1-9-]+[-]npm[.]pkg[.]dev\/.*\/:username=oauth2accesstoken)/g;
+  fromConfigLines = fromConfigLines.replace(legacyRegex, `$1:_authToken=${creds}`)
+
+  // Read configs from project npmrc file. For each:
+  // - registry config, create an auth token config in the user npmrc file (expect an auth token or password config already exists)
+  // - auth token config, print a warning and remove it.
+  // - password config, print a warning and move it to the user npmrc file.
+  // - everything else, keep it in the project npmrc file.
+  for (const line of fromConfigLines.split('\n')) {
+    let config = c.parseConfig(line);
+    switch (config.type) {
+      case c.configType.Registry:
+        fromConfigs.push(config);
+        registryAuthConfigs.set(config.registry, new c.AuthTokenConfig(config.registry, creds));
+        break;
+      case c.configType.AuthToken:
+        console.warn(`Found an auth token for the registry ${config.registry} in the project npmrc file. Moving it to the user npmrc file...`);
+        break;
+      case c.configType.Password:
+        console.warn(`Found password for the registry ${config.registry} in the project npmrc file. Moving it to the user npmrc file...`);
+        registryAuthConfigs.set(config.registry, config);
+        break;
+      default:
+        fromConfigs.push(config);
+    }
+  }
+
+  if (fs.existsSync(toConfigPath)) {
+    const toConfigLines = await fs.promises.readFile(toConfigPath, "utf8")
+
+    // refresh tokens for all auth token configs; keep everything else unchanged.
+    for (const line of toConfigLines.split('\n')) {
```

Comment:
I was originally splitting it by `os.EOL` but it would cause the tests to fail on windows. It seems the multi-line string pattern with backticks are joined by `\n` only even on windows. Not sure what's the best practice here.

Reply:
Done.